### PR TITLE
feat (Reject trip request) #181414569

### PR DESCRIPTION
**Pull request by NNesta**
# 1. Name of task
- Reject Request (By Requester’s Manager) UI
# 2. Pivot Tracker ID
-[ #181414564](https://www.pivotaltracker.com/story/show/181414569)

# 3. Branch Name
- ft-Reject-Request-##181414569
# 4. Description of task
In some cases, travel requests will not be allowed to proceed to the next phase

**As a manager**, I should be able to reject a request created by my direct report
**So that**, my direct report’s travel request can be aborted

## **Acceptance Criteria**
- Manager is able to reject request
- Status of the approved request changes to ‘Rejected’
- Request owner can see the status of the rejected request as ‘Rejected’
# 5. How can PR be tested.
Given that the user is logged in,
- (/dashboard) click on more button then will rediredt you to /approval page where you can get button of rejecting
** User Interface on the page**
Screen Shot 2022-05-19 at 09 25 02

### CODE REVIEW

**Review by mucyoalexaxel — DISMISSED** [review]
@NNesta  remove `src/.DS_Store`

**Comment by NNesta** [comment]
> @NNesta remove `src/.DS_Store`

That is what I did

File: test/mocks/handlers/approve.handler.js [hunk 1]
```diff
@@ -4,25 +4,36 @@ import { localUrl } from '../../../src/api';
...
+const userId2 = '861d7c9d-a8a8-4308-bce8-2c6c6a66c840';
 
 // eslint-disable-next-line import/prefer-default-export
 export const approveHandlers = [
   rest.get(`${baseURl}v1/user/profile/${userId}`, (req, res, ctx) => {
     return res(ctx.status(201), ctx.json(profile), ctx.delay(100));
   }),
+  rest.get(`${baseURl}v1/user/profile/${userId2}`, (req, res, ctx) => {
+    return res(
+      ctx.status(201),
```
Comment (EmyRukundo) [inline test/mocks/handlers/approve.handler.js]: wrong status code

**Review by EmyRukundo — CHANGES_REQUESTED** [review]

File: src/components/tables/ApprovalComponent.js [hunk 2]
```diff
@@ -179,33 +200,40 @@ function ApprovalComponent(props) {
...
+        (profileLoading ? (
+          <Spinner />
+        ) : (
+          <div className="bg-white px-8 md:px-48 h-full opacity-100 absolute inset-0 flex justify-center z-30">
+            <ProfileModal data={profile} setModalOn={setModalOn} />
+          </div>
+        ))}
     </div>
   );
 }
```
Comment (SergeNm) [inline src/components/tables/ApprovalComponent.js:248]: prefer refactoring this component

File: test/mocks/handlers/approve.handler.js (hunk 1 above)
Comment (NNesta) [reply test/mocks/handlers/approve.handler.js]: Correected

Comment (NNesta) [reply test/mocks/handlers/approve.handler.js]: Correected

File: test/integration/tripApprove.test.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
```
Comment (SergeNm) [inline test/integration/tripApprove.test.js:1]: you can prefer to mock the response in the handlers for testing

File: test/integration/tripApprove.test.js [hunk 4]
```diff
@@ -25,32 +26,75 @@ const state = {
...
+      expect(await findByText('Close')).toBeInTheDocument();
+      const close = screen.getByTestId('close-button');
+      const gender = screen.getByText('Male');
+      expect(close).toBeInTheDocument();
+      expect(gender).toBeInTheDocument();
+      fireEvent.click(screen.getByTestId('close-button'));
+    }, 1500);
+  });
+});
+/* describe('Approve or reject atrip request', () => {
```
Comment (SergeNm) [inline test/integration/tripApprove.test.js]: remove these comments

**Review by SergeNm — DISMISSED** [review]

**Review by SergeNm — COMMENTED** [review]
It's a good work otherwise

File: test/mocks/handlers/approve.handler.js [hunk 5]
```diff
@@ -4,24 +4,35 @@ import { localUrl } from '../../../src/api';
...
 
 // eslint-disable-next-line import/prefer-default-export
 export const approveHandlers = [
   rest.get(`${baseURl}v1/user/profile/${userId}`, (req, res, ctx) => {
-    return res(ctx.status(201), ctx.json(profile), ctx.delay(100));
+    return res(ctx.status(200), ctx.json(profile), ctx.delay(100));
+  }),
+  rest.get(`${baseURl}v1/user/profile/${userId2}`, (req, res, ctx) => {
+    return res(
+      ctx.status(200),
```
Comment (EmyRukundo) [inline test/mocks/handlers/approve.handler.js:32]: status code is wrong

**Review by EmyRukundo — DISMISSED** [review]

**Review by mucyoalexaxel — DISMISSED** [review]
LGTM.

**Review by iyaremye-faustin — DISMISSED** [review]
Working fine

**Review by iyaremye-faustin — APPROVED** [review]
Working fine

**Review by mucyoalexaxel — APPROVED** [review]
LGTM

**Pull request merged**